toolchain/parse/handle_impl.cpp

```diff
@@ -64,17 +63,16 @@ auto HandleImplAfterForall(Carbon::Parse::Context& context) -> void {
 
 auto HandleImplBeforeAs(Carbon::Parse::Context& context) -> void {
   auto state = context.PopState();
-  if (state.has_error) {
-    context.ReturnErrorOnState();
-    return;
-  }
   if (auto as = context.ConsumeIf(Lex::TokenKind::As)) {
-    context.AddLeafNode(NodeKind::ImplAs, *as);
+    context.AddNode(NodeKind::TypeImplAs, *as, state.subtree_start,
+                    state.has_error);
```

Comment:
It looks like we don't have any test coverage for the case where there's a parse error before `as`, where we'd be building a subtree with `has_error == true` here. I *think* this is handling that properly, but it'd be good to have a test case to cover that.